app/services/MeansOfTransportIdentificationTypesActiveService.scala

```diff
@@ -33,6 +34,14 @@ class MeansOfTransportIdentificationTypesActiveService @Inject() (referenceDataC
   ): Future[Seq[Identification]] =
     referenceDataConnector.getMeansOfTransportIdentificationTypesActive().map(filter(_, index, borderModeOfTransport)).map(sort)
 
+  def getBorderMeansIdentification(code: String)(implicit hc: HeaderCarrier): Future[Identification] =
+    for {
+      qualifiers <- referenceDataConnector.getMeansOfTransportIdentificationTypesActive()
+      qualifierFound = qualifiers
+        .find(_.code == code)
+        .getOrElse(referenceDataException("borderMeansIdentification", code, qualifiers))
```

Comment:
Could we instead add a method to the connector to user `/filtered-lists` for the `TypeOfIdentificationofMeansOfTransportActive` code list? The filter gets passed through all the way to the mongo query - more efficient than handling the entire list in memory.

The method would still return a Seq[Identification]. Perhaps we could refactor to return NonEmptyList / NonEmptySeq? Then in the service method we can safely call `.head` without having to do a `.getOrElse`

Reply:
```
  implicit def responseHandlerGeneric[A](implicit reads: Reads[A]): HttpReads[NonEmptySeq[A]] =
    (_: String, _: String, response: HttpResponse) => {
      response.status match {
        case OK =>
          (response.json \ "data").validate[Seq[A]] match {
            case JsSuccess(Nil, _) =>
              throw new NoReferenceDataFoundException
            case JsSuccess(head :: tail, _) =>
              NonEmptySeq(head, tail)
            case JsError(errors) =>
              throw JsResultException(errors)
          }
        case e =>
          logger.warn(s"[ReferenceDataConnector][responseHandlerGeneric] Reference data call returned $e")
          throw new Exception(s"[ReferenceDataConnector][responseHandlerGeneric] $e - ${response.body}")
      }
    }
```